pkg/network/telemetry/metric.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package telemetry
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"go.uber.org/atomic"
+)
+
+// Metric represents a named piece of telemetry
+type Metric struct {
+	name  string
+	tags  []string
+	opts  []string
+	value *atomic.Int64
+
+	// metrics of type OptMonotonic use this value
+	// when Delta() is called
+	prevValue *atomic.Int64
+}
+
+// NewMetric returns a new `Metric` instance
+func NewMetric(name string, tagsAndOptions ...string) *Metric {
+	tags, opts := splitTagsAndOptions(tagsAndOptions)
+	m := &Metric{
+		name:  name,
+		value: atomic.NewInt64(0),
+		tags:  tags,
+		opts:  opts,
+	}
+
+	if contains(OptMonotonic, m.opts) {
+		m.prevValue = atomic.NewInt64(0)
+	}
+
+	r.Lock()
```

Comment:
where did `r` come from?

Reply:
it's a global metric registry defined here: https://github.com/DataDog/datadog-agent/blob/fc9ee0fa53ebc9b5663dd41b94db58bf25393856/pkg/network/telemetry/repository.go#L12
I'll rename it to `globalRegistry` to make it more clear